### ISSUE TITLE
Unclear OAuth2 error logs off NextCloud

### ISSUE THREAD

**Issue by maherbm** [nextcloud/server#11872]
### Steps to reproduce
1. Install NextCloud 13.0.6snap1 using snap
2. Install Rocket Chat 0.69.2 using snap (on the same system)
3. add Rocket Chat as an OAuth 2.0 client
4. Configure Rocket Chat to use NextCloud for OAuth authentication
5. Log in on NextCloud using browser
6. Go to Rocket Chat and log in using NextCloud as OAuth provider
7. System suddenly logs off the NextCloud session. However, Rocket Chat stays authenticated

### Expected behaviour
To stay logged in on NextCloud after using NextCloud OAuth provider to authenticate on Rocket Chat

### Actual behaviour
The system suddenly logs off the NextCloud session after using NextCloud as OAuth provider. However, Rocket Chat stays logged in.
The log shows an OAuth related error, but not much details are provided. Hereby the error with the OAuth Client Identifier replaced with ***

```
{"reqId":"P40fvHMkdxGleulTX4sk","level":2,"time":"2018-10-15T13:21:27+00:00","remoteAddr":"::1","user":"--","app":"core","method":"POST","url":"\/index.php\/apps\/oauth2\/api\/v1\/token","message":"Login failed: '*********************************' (Remote IP: '::1')","userAgent":"Meteor\/METEOR@1.6.1.3","version":"13.0.6.1"} 
```

### Server configuration

**Operating system**: Ubuntu 16.04

**Nextcloud version:** 13.0.6

**Updated from an older Nextcloud/ownCloud or fresh install: Automatic update using snap for the last 6 months. However, integration of Rocket Chat was not tested with previous versions

**Where did you install Nextcloud from:** snap

**Signing status:**
<details>
<summary>Signing status</summary>

```
No errors have been found.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - activity: 2.6.1
  - apporder: 0.5.0
  - bruteforcesettings: 1.1.0
  - calendar: 1.6.1
  - caniupdate: 0.1.3
  - comments: 1.3.0
  - dav: 1.4.7
  - deck: 0.4.1
  - drawio: 0.9.0
  - external: 3.0.4
  - federatedfilesharing: 1.3.1
  - federation: 1.3.0
  - files: 1.8.0
  - files_markdown: 2.0.4
  - files_pdfviewer: 1.2.1
  - files_rightclick: 0.8.4
  - files_sharing: 1.5.0
  - files_texteditor: 2.5.1
  - files_trashbin: 1.3.0
  - files_versions: 1.6.0
  - files_videoplayer: 1.2.0
  - firstrunwizard: 2.2.1
  - impersonate: 1.0.4
  - logreader: 2.0.0
  - lookup_server_connector: 1.1.0
  - nextcloud_announcements: 1.2.0
  - notifications: 2.1.2
  - oauth2: 1.1.1
  - onlyoffice: 2.0.4
  - password_policy: 1.3.0
  - provisioning_api: 1.3.0
  - quota_warning: 1.2.0
  - ransomware_protection: 1.1.2
  - serverinfo: 1.3.0
  - sharebymail: 1.3.0
  - spreed: 3.2.5
  - systemtags: 1.3.0
  - theming: 1.4.5
  - twofactor_backupcodes: 1.2.3
  - workflowengine: 1.3.0
Disabled:
  - admin_audit
  - encryption
  - files_external
  - gallery
  - richdocuments
  - survey_client
  - user_external
  - user_ldap
  - user_saml
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
{
    "system": {
        "apps_paths": [
            {
                "path": "\/snap\/nextcloud\/current\/htdocs\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/var\/snap\/nextcloud\/current\/nextcloud\/extra-apps",
                "url": "\/extra-apps",
                "writable": true
            }
        ],
        "supportedDatabases": [
            "mysql"
        ],
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "memcache.local": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0
        },
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "xxxxxxxxx"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "http:\/\/xxxxxxxxxx",
        "dbtype": "mysql",
        "version": "13.0.6.1",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "skeletondirectory": "",
        "remember_login_cookie_lifetime": 31536000,
        "maintenance": false,
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_smtpauthtype": "LOGIN",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "25",
        "data-fingerprint": "5aff4629f5166bdd4e0cc0d588af672d",
        "loglevel": 2
    }
}
```
</details>

**Are you using external storage, if yes which one:** No

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** No

### Client configuration
**Browser:** Chrome Version 69.0.3497.100 (Official Build) (64-bit)

**Operating system:** Windows 10

#### Web server error log
<details>
<summary>Web server error log</summary>

```
No new enteries to the apache or php error logs
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
{"reqId":"P40fvHMkdxGleulTX4sk","level":2,"time":"2018-10-15T13:21:27+00:00","remoteAddr":"::1","user":"--","app":"core","method":"POST","url":"\/index.php\/apps\/oauth2\/api\/v1\/token","message":"Login failed: '*********************************' (Remote IP: '::1')","userAgent":"Meteor\/METEOR@1.6.1.3","version":"13.0.6.1"} 
```
</details>

**Comment by maherbm** [comment]
The issues  generated by the bot are not related to the problem I mentioned

**Comment by maherbm** [comment]
I decreased the level of logging and found out the following error occurred directly after the OAuth error:

```
{"reqId":"lUT6HyEaskH1oG4B5gk3","level":1,"time":"2018-10-19T14:56:11+00:00","remoteAddr":"::1","user":"--","app":"core","method":"POST","url":"\/index.php\/apps\/oauth2\/api\/v1\/token","message":"Bruteforce attempt from \"::1\" detected for action \"login\".","userAgent":"Meteor\/METEOR@1.6.1.3","version":"13.0.6.1"}
```

It seems the Bruteforce mechanism kicks the user out  after the OAuth error. I have removed the bruthforce app from nextcloud, but it did not change anything. It seems it comes from another bruteforce mechanism inside the core of nextcloud.

On one hand, the NextCloud as an OAuth providers works well for RocketChat and the user stays logged in at RocketChat. However, for some reason NextCloud thinks the OAuth authentication failed and it causes an error followed up by a brute force defence.
Maybe RocketChat does not return the proper feedback back to NextCloud. Hereby a screenshot of my OAuth settings in RocketChat:

![rocketchat-oauth](https://user-images.githubusercontent.com/4059100/47229191-7209d600-d3c7-11e8-9618-5b9111fdee5c.png)
